UnityProject/Assets/Scripts/Systems/StatusesAndEffects/Implementations/SpeedBuff.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using Logs;
+using Player.Movement;
+using Systems.StatusesAndEffects.Interfaces;
+using UnityEngine;
+
+namespace Systems.StatusesAndEffects.Implementations
+{
+	[CreateAssetMenu(fileName = "Speed Buff", menuName = "ScriptableObjects/StatusEffects/SpeedBuff")]
+	public class SpeedBuff : StatusEffect, IExpirableStatus, IMovementEffect, IStackableStatus
```

Comment:
imo Would have been nice if it was checking for the nicotine reagent in the blood and then did stuff from that?

Reply:
but its already doing that?